### ISSUE TITLE
`export foo from './foo'` is throwing error about "Node must be provided when reporting error if location is not provided"

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#7554]
**Tell us about your environment**

* **ESLint Version:**

3.9.1

* **Node Version:**

v6.9.1

* **npm Version:**

3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

https://github.com/gajus/eslint-config-canonical

**What did you do? Please include the actual source code causing the issue.**

```js
export foo from './foo';

```

**What did you expect to happen?**

The validation to pass the test.

**What actually happened? Please include the actual, raw output from ESLint.**

Error:

```
Node must be provided when reporting error if location is not provided
AssertionError: Node must be provided when reporting error if location is not provided
    at EventEmitter.module.exports.api.report (/test/node_modules/eslint/lib/eslint.js:970:20)
    at RuleContext.report (/test/node_modules/eslint/lib/rule-context.js:144:21)
    at EventEmitter.Program:exit (/test/node_modules/eslint-plugin-import/lib/rules/prefer-default-export.js:69:19)
    at emitOne (events.js:101:20)
    at EventEmitter.emit (events.js:188:7)
    at NodeEventGenerator.leaveNode (/test/node_modules/eslint/lib/util/node-event-generator.js:49:22)
    at CodePathAnalyzer.leaveNode (/test/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:628:23)
    at CommentEventGenerator.leaveNode (/test/node_modules/eslint/lib/util/comment-event-generator.js:110:23)
    at Controller.leave (/test/node_modules/eslint/lib/eslint.js:930:36)
    at Controller.__execute (/test/node_modules/estraverse/estraverse.js:397:31)
```

**Other information**

This used to work fine 24 hours ago. Now all related builds started failing at once. I cannot see any recent releases of eslint or babel-eslint. I can therefore only assume that some sub-dependency is causing this behaviour.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

However, this is an issue with `eslint-plugin-import`, not with ESLint itself. I would recommend following [the issue on that repo](https://github.com/benmosher/eslint-plugin-import/issues/653) instead.

Also see: https://github.com/eslint/eslint/issues/7544

**Comment by gajus** [comment]
You are right. Sorry. I can confirm that downgrading `eslint-plugin-import` to v2.0.0 fixes the issue.
